[React quickstart] removes the possibility of returning false

Jay-study-nildana opened the pull request:
The code in its current form wont work because it might return 'false'. I have updated the code so that a JSX element is returned no matter what is the status of authentication.

<!---
Pull Requests for Quickstart Guides can still be submitted here, but most other documentation content is no longer hosted on GitHub and therefore no longer open-sourced. If you are an Auth0 employee trying to make a change, please [submit a ticket](https://auth0team.atlassian.net/servicedesk/customer/portal/9). Thank you!
--->

Jay-study-nildana committed:
removes the possibility of returning false

The code in its current form wont work because it might return 'false'. I have updated the code so that a JSX element is returned no matter what is the status of authentication.

stevehobbsdev commented on articles/quickstart/spa/react/01-login.md:
```diff
@@ -126,25 +126,32 @@ The Auth0 React SDK helps you retrieve the [profile information](https://auth0.c
 import { useAuth0 } from "@auth0/auth0-react";
 
-const Profile = () => {
+const UserProfile = () => {
```
What's the reason for the component name change?

I'm ok with it but there are references to the old name in the surrounding text, which must also be updated.

Jay-study-nildana replied:
What's the reason for the component name change?

No reason at all. I have no specific preference for it.

Jay-study-nildana replied:
I just realized that was the name I gave for my own custom component, he he :P It's funny. 

Please keep it how it was before. Dont change the name. Let me fix that.

Jay-study-nildana replied:
Alright, I have put back the old component change. My bad again. I should not have put it in the first place (sorry, I just started learning react only a few weeks ago).

But, I am using Auth0 on my API server, so need to use it with my react learning process. 

Thank you.

frederikprijck replied:
This still isn't changed. You only changed line 153 back to the original value.

Jay-study-nildana replied:
@stevehobbsdev @frederikprijck 

Folks, 

I am really new to react. any way you wish to resolve this, please do. Really, I don't mind.

Do, what is best but yeah, stop it from returning false is all I ask. If you can. 

Thank you.

stevehobbsdev replied:
Thanks @Jay-study-nildana, you mentioned you are new to React and we understand the spirit of the change you're trying to make, so we can take it from here and change the snippet.

Let's go with [the suggestion here](https://github.com/auth0/docs/pull/9548#discussion_r558601008). @frederikprijck can you please modify the PR to make this change when you have some time? If you're unable to change the PR, please raise a new one and link it here.

stevehobbsdev reviewed (changes requested):
Thanks for the contribution! If we're making this change here, you probably also want to make this change in the ["Calling an API" chapter](https://auth0.com/docs/quickstart/spa/react/02-calling-an-api) which has a similar code snippet and modifies this `Profile` component.

I left a remark about the changing of the component name too - not sure it's totally necessary but happy to go with it as long as all the text is covered.

Jay-study-nildana committed:
reverted an accidental change of code.

I just realized that this was the component name for my own project. This line should be as it was before.

frederikprijck commented on articles/quickstart/spa/react/01-login.md:
```diff
@@ -126,25 +126,32 @@ The Auth0 React SDK helps you retrieve the [profile information](https://auth0.c
 
-  return (
-    isAuthenticated && (
+  if (isAuthenticated) {
```
Personally I am more a fan of showing:

```
return (
  isAuthenticated ? (
    <div>
      <img src={user.picture} alt={user.name} />
      <h2>{user.name}</h2>
      <p>{user.email}</p>
    </div>
  ) : ''
);
```

What do you think about this @stevehobbsdev and @Jay-study-nildana ? This avoids returning false without adding additional markup.

Jay-study-nildana replied:
@frederikprijck @stevehobbsdev 

See, there you go. I cannot do these nifty little pieces of code yet :P I like this one better.  It looks better too.

frederikprijck committed:
Update based on Review comments

frederikprijck commented:
Thanks for PR, we are grateful for any improvements made to our documentation.

In this case, we will be closing this PR because the current docs are actually aligned with React Best practises: https://reactjs.org/docs/jsx-in-depth.html#booleans-null-and-undefined-are-ignored

In the above link, it states that `false`, `null`, `undefined`, and `true` are valid children. They simply don’t render.

Sorry for the confusion!
